### PR TITLE
Adds -Wcast-qual, -Wmissing-field-initializers warnings flags.

### PR DESCRIPTION
**Pull request by mookerji**
Adds warnings flags for unsafe cast, incomplete struct initialization.

/cc @yuzheyan @mandeepsandhu

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -63,6 +63,9 @@ endif()
 
 function (cxx_add_warning_flags target)
   target_add_compile_flags_if_supported(${target} PRIVATE -Wall -Wextra)
+  target_add_compile_flags_if_supported(${target} PRIVATE
+    -Wcast-qual
+    -Wmissing-field-initializers)
```
Comment (mandeepsandhu) [inline src/CMakeLists.txt]: This option is enabled by `-Wextra` I think. Can you check if its true for the version of `gcc` we're using?

Comment (mookerji) [reply src/CMakeLists.txt]: Oh shoot, it is.

Comment (mookerji) [reply src/CMakeLists.txt]: Nice catch!

Comment (mookerji) [reply src/CMakeLists.txt]: Removed

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -62,10 +62,10 @@ if (ENABLE_COMPILER_WARNINGS)
 endif()
 
 function (cxx_add_warning_flags target)
-  target_add_compile_flags_if_supported(${target} PRIVATE -Wall -Wextra)
   target_add_compile_flags_if_supported(${target} PRIVATE
-    -Wcast-qual
-    -Wmissing-field-initializers)
+    -Wall
+    -Wextra
+    -Wcast-qual)
```
Comment (mandeepsandhu) [inline src/CMakeLists.txt]: Curious about how you're determining which warning flags to enable?

LGTM otherwise.

**Comment by mookerji** [comment]
Closing this for the time being.

**Pull request closed without merging**